### PR TITLE
[Refactoring] Move TypeInfo to a separate module

### PR DESCRIPTION
**Pull request by elazarg**
It seems appropriate, since `TypeInfo` is not a syntactic node.

### CODE REVIEW

File: mypy/nodes.py [hunk 1]
```diff
@@ -1917,301 +1931,6 @@ def accept(self, visitor: ExpressionVisitor[T]) -> T:
...
-
-    # If this class is defined by the TypedDict type constructor,
-    # then this is not None.
-    typeddict_type = None  # type: Optional[mypy.types.TypedDictType]
-
-    # Is this a newtype type?
-    is_newtype = False
-
-    # Alternative to fullname() for 'anonymous' classes.
-    alt_fullname = None  # type: Optional[str]
```
Comment (elazarg) [inline mypy/nodes.py]: This field is removed. It is never set to anything other than `None`. If this PR is rejected, I can post a separate one just for this removal.

**Comment by gvanrossum** [comment]
Not sure what happened to alt_fullname, IIRC I added it, maybe the use case vanished. Feel free to submit its removal as a separate (small) PR, the big refactor will take more time to review (and TBH it's at the end of the queue).

**Comment by elazarg** [comment]
Will do.
I assumed it's not going to be top priority (although there shouldn't be anything serious here other than moving code around; I wish there was an easy way to verify that mechanically).

**Comment by gvanrossum** [comment]
This really scares me because committing this will cause merge conflicts in every other PR. I don't know how to handle such refactors TBH. When you move code to a different file how can you be sure that it's not accidentally changed in some subtle way?

**Comment by elazarg** [comment]
I can reduce some of the churn by not moving `set_original_def` and other methods to TypeInfo, and by exporting TypeInfo from `nodes.py`. Then I think most of the change is a simple copy-paste.

If it's still too big, I should probably close this PR.

**Comment by elazarg** [comment]
OK. This is not so easy.

**Pull request closed without merging**